src/ChainSafe.Gaming.SygmaClient/SygmaClient.cs

```diff
@@ -120,7 +123,56 @@ public async Task<TransactionRequest> BuildApprovals<T>(Transfer<T> transfer, Ev
         public Task<TransactionRequest> BuildTransferTransaction<T>(Transfer<T> transfer, EvmFee fee)
             where T : TransferType
         {
-            throw new System.NotImplementedException();
+            switch (transfer.Resource.Type)
+            {
+                case ResourceType.NonFungible:
+                    var nonFungible = transfer as Transfer<NonFungible>;
+                    return Erc721Transfer(
+                        nonFungible!.Details.TokenId,
+                        nonFungible.Details.Recipient,
+                        nonFungible.To.Id.ToString(),
+                        transfer.Resource.ResourceId,
+                        fee);
+                default:
+                    throw new NotImplementedException($"This type {transfer.Resource.Type} is not implemented yet");
+            }
+        }
+
+        private async Task<TransactionRequest> Erc721Transfer(string tokenId, string recipientAddress, string domainId, string resourceId, EvmFee feeData)
+        {
+            var sourceDomainConfig = clientConfiguration.SourceDomainConfig();
+            var depositData = CreateDepositData(tokenId, recipientAddress);
+            var bridge = new Bridge(contractBuilder, sourceDomainConfig.Bridge);
+
+            var tx = await bridge.Contract.PrepareTransactionRequest(
+                "deposit",
+#pragma warning disable SA1118
+                new object[]
+                {
+                domainId,
+                resourceId,
+                depositData,
+                feeData.FeeData ?? "0x0",
+                });
+#pragma warning restore SA1118
+            return tx;
+        }
+
+        // In sygmas SDK we also have Substrate, (check helpers.ts -> CreateERCDepositData)
+        // We don't need paraChainID just yet since we are only working with Ethereum
+        private string CreateDepositData(string tokenId, string recipient)
+        {
+            List<byte> list = new List<byte>();
+            list.AddRange(BigInteger.Parse(tokenId).ToByteArray());
```

Comment:
Not sure about this one.

In the original, we pad both hexes by zeros to the length of 32
https://github.com/sygmaprotocol/sygma-sdk/blob/bd952310638224af842e9620679ee606f9b55ea1/packages/sdk/src/chains/EVM/helpers.ts#L53

Here are the docs on `hexZeroPad`
https://docs.ethers.org/v5/api/utils/bytes/#utils-hexZeroPad

Reply:
Fixed.